sage-ios/SAGE/SAGE/SchoolDetailViewController.swift

```diff
@@ -216,8 +216,12 @@ class SchoolDetailViewController: SGTableViewController {
         } else {
             userProfile = self.school!.students![indexPath.row]
         }
-        let viewController = ProfileViewController(user: userProfile)
-        self.navigationController!.pushViewController(viewController, animated: true)
+        if let role = LoginOperations.getUser()?.role {
```

Comment:
can you make this condition wrap the entire logic? so basically this whole method is encapsulated by the condition